[SQL] Fix error message for unimplemented feature

The following SQL for an unimplemented feature(UNNEST with two arguments):
`SELECT * FROM UNNEST(ARRAY [1, 2, 3, 4, 5], ARRAY[3, 2, 1]);`

Produces the following Rust error:
**Rust error: the Rust code generated based on the SQL (possibly combined with any user-provided UDF Rust and TOML code) failed to compile.
This should not happen (except if the error is in user-provided UDF code).
Please file a bug report with the example SQL and any UDF Rust/TOML that triggers it at:
error[E0308]: mismatched types**

--- a/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java
@@ -702,6 +702,9 @@ public class CalciteToDBSPCompiler extends RelVisitor
             DBSPTypeTuple pair = type.to(DBSPTypeTuple.class);
             indexType = pair.getFieldType(1);
         }
+        if (inputRowType.size() > 1) {
+            throw new UnimplementedException("UNNEST with multiple vectors", node);
+        }
         DBSPVariablePath data = new DBSPVariablePath(inputRowType.ref());
         DBSPType arrayElementType = data.deref().field(0).getType();
         DBSPClosureExpression getField0 = data.deref().field(0).closure(data);

--- a/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/RegressionTests.java
+++ b/sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/RegressionTests.java
@@ -1864,6 +1864,13 @@ public class RegressionTests extends SqlIoTest {
     }
 
     @Test
+    public void issue3547() {
+        this.queryFailingInCompilation(
+                "SELECT * FROM UNNEST(ARRAY [1, 2, 3, 4, 5], ARRAY[3, 2, 1]);",
+                "UNNEST with multiple vectors");
+    }
+
+    @Test
     public void issue2736() {
         var ccs = this.getCCS("""
                 create table latest_cells(

